File: Programmers/Lv2/Solution_12945.java
package Lv2;

public class Solution_12945 {
    public int solution(int n) {
        long[] dp = new long[n + 1];

        dp[0] = 0L;
        dp[1] = 1L;

        for(int i = 2; i <= n; i++) {
            dp[i] = ( dp[i - 1] + dp[i - 2] ) % 1234567;
        }

        return (int) dp[n];
    }
}

File: Programmers/Lv2/Solution_12945.cpp
#include <vector>

using namespace std;

int solution(int n) {
    vector<int> dp;
    dp.push_back(0);
    dp.push_back(1);

    for(int i = 2; i <= n; i++) {
        int cur = (dp[i - 1] + dp[i - 2]) % 1234567;
        dp.push_back(cur);
    }

    return dp[n];
}
